apps/gitness/src/pages-v2/pull-request/pull-request-list.tsx

```diff
@@ -75,6 +82,12 @@ export default function PullRequestListPage() {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [page, queryPage, setPage])
 
+  useEffect(() => {
+    if (searchParams.get('label_by')) {
+      setPopulateLabelStore(true)
+    }
+  }, [searchParams])
```

Comment:
This `useEffect` may cause unnecessary re-renders even if the actual parameters haven’t changed, because the searchParams object itself might be re-instantiated every render, making it a new reference each time.

Prefer below:

```
const labelBy = searchParams.get('label_by');

useEffect(() => {
  if (labelBy) {
    setPopulateLabelStore(true);
  }
}, [labelBy]);
```